src/middlewared/middlewared/plugins/jbof/redfish/client.py

```diff
@@ -0,0 +1,300 @@
+import enum
+import json
+import logging
+import requests
+import urllib3
+
+from urllib.parse import urlencode
+
+DEFAULT_REDFISH_TIMEOUT_SECS = 10
+HEADER = {'Content-Type': 'application/json', 'Vary': 'accept'}
+REDFISH_ROOT_PATH = '/redfish/v1'
+ODATA_ID = '@odata.id'
+
+logger = logging.getLogger(__name__)
+
+
+class InvalidCredentialsError(Exception):
+    pass
+
+
+class AuthMethod(enum.Enum):
+    BASIC = 'basic'
+    SESSION = 'session'
+
+    def choices():
+        return [x.value for x in AuthMethod]
+
+
+class RedfishClient:
+
+    client_cache = {}
+
+    @classmethod
+    def setup(cls):
+        # Silence InsecureRequestWarning: Unverified HTTPS request is being made to host
+        requests.packages.urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+    @classmethod
+    def ping(cls, mgmt_ip, timeout=DEFAULT_REDFISH_TIMEOUT_SECS):
+        mgmt_ip = mgmt_ip.rstrip('/')
+        url = f'https://{mgmt_ip}{REDFISH_ROOT_PATH}'
+        r = requests.get(url, verify=False, timeout=timeout)
```

Comment:
Is there a reason we're not checking the HTTP status code `r.status_code` here? Should we log an error message if this fails?